synapse/visibility.py

```diff
@@ -301,9 +308,17 @@ def _check_client_allowed_to_see_event(
             _check_filter_send_to_client(event, clock, retention_policy, sender_ignored)
             == _CheckFilter.DENIED
         ):
+            logger.debug(
+                "_check_client_allowed_to_see_event(event=%s): Filtered out event because `_check_filter_send_to_client` returned `_CheckFilter.DENIED`",
+                event.event_id,
+            )
             return None
 
     if event.event_id in always_include_ids:
+        logger.debug(
+            "_check_client_allowed_to_see_event(event=%s): Returning event because it's included in the `always_include_ids`",
+            event.event_id,
+        )
```

Comment:
To have less noise, maybe we only want to log when an event is filtered out and not why it was returned unscathed.